### PR TITLE
test: add Jest unit tests for MainLayoutContainer

### PR DESCRIPTION
**Pull request by arbisoft-qaisarirfan**
#### **Summary**  
This PR adds **comprehensive Jest unit tests** for the `MainLayoutContainer` component to verify its rendering behavior, conditional logic, and edge cases.  

---

### **Changes Implemented**  
✅ **Added Jest test suite for `MainLayoutContainer`.**  
✅ **Verified correct component rendering** with expected UI elements.  
✅ **Tested conditional rendering logic** based on props and state.  
✅ **Handled edge cases and error scenarios** to ensure robustness.  
✅ **Achieved high test coverage** for `MainLayoutContainer`.  

### **Impact**  
- **Ensures the correctness of `MainLayoutContainer`.**  
- **Reduces risk of regressions with future changes.**  
- **Improves overall test coverage and project stability.**  

---

🔗 **Reviewers:** _(@farhan2742 @arslanather @wkhaliddev1 @sameeramin)_  

https://projects.arbisoft.com/project/arbisoft-sessions-portal-20/us/78

Screenshot 2025-02-14 at 12 41 17 PM
Screenshot 2025-02-14 at 12 41 35 PM

screencapture-localhost-6006-2025-02-14-12_46_43

✅ **Ready for review!** 🚀

### CODE REVIEW

File: src/components/Sidebar/styled.tsx [hunk 1]
```diff
@@ -24,15 +26,16 @@ export const MenuStack = styled(Stack, {
 
 export const MenuItem = styled(Stack, {
   name: "MenuItem",
-})<{ isSelected: boolean }>((props) => {
+  shouldForwardProp,
+})<{ $isSelected: boolean }>((props) => {
```
Comment (ahmad-arbisoft) [inline src/components/Sidebar/styled.tsx:30]: Can we destructure the `props` here?

File: src/components/containers/MainLayoutContainer/mainLayoutContainer.tsx [hunk 2]
```diff
@@ -17,14 +17,15 @@ const MainLayoutContainer = ({ children, rightSidebar }: TMainLayoutContainer) =
...
       <Navbar />
       <MainContainer maxWidth="xl">
-        <LeftSidebar>
+        <LeftSidebar data-testid="left-sidebar">
           <Sidebar />
         </LeftSidebar>
         <ContentContainer container>{children}</ContentContainer>
         {isValidElement(rightSidebar) && (
           <RightSidebar
             // Hide on small screens
```
Comment (ahmad-arbisoft) [inline src/components/containers/MainLayoutContainer/mainLayoutContainer.tsx:26]: Unnecessary comment here

**Review by ahmad-arbisoft — APPROVED** [review]
LGTM

**Review by farhan2742 — APPROVED** [review]
LGTM
